znft/keystore.go

```diff
@@ -0,0 +1,147 @@
+	"gopkg.in/yaml.v2"
+)
+
+// ListStorageAccounts List available accounts
+func ListStorageAccounts(homedir string) []common.Address {
+	am := getAccountManager(homedir)
+	addresses := am.Accounts()
+
+	return addresses
+}
+
+// DeleteAccount deletes account from wallet
+func DeleteAccount(homedir, address string) bool {
+	am := getAccountManager(homedir)
+
+	wallet, err := am.Find(accounts.Account{
+		Address: common.HexToAddress(address),
+	})
+
+	if err != nil && wallet == nil {
+		fmt.Printf("failed to find account %s, error: %s", address, err)
+		return false
+	}
+
+	return true
+}
+
+func getAccountManager(homedir string) *accounts.Manager {
+	keyDir := path.Join(homedir, WalletDir)
+	ks := keystore.NewKeyStore(keyDir, keystore.StandardScryptN, keystore.StandardScryptP)
+	config := &accounts.Config{InsecureUnlockAllowed: false}
+	am := accounts.NewManager(config, ks)
+	return am
+}
+
+// AccountExists checks if account exists
+func AccountExists(homedir, address string) bool {
+	am := getAccountManager(homedir)
+
+	wallet, err := am.Find(accounts.Account{
+		Address: common.HexToAddress(address),
+	})
+
+	if err != nil && wallet == nil {
+		fmt.Printf("failed to find account %s, error: %s\n", address, err)
+		return false
+	}
+
+	status, _ := wallet.Status()
+	url := wallet.URL()
+
+	fmt.Printf("Account exists. Status: %s, Path: %s\n", status, url)
+
+	return true
+}
+
+// CreateKeyStorage create, restore or unlock key storage
+func CreateKeyStorage(homedir, password string) error {
+	keyDir := path.Join(homedir, WalletDir)
+	ks := keystore.NewKeyStore(keyDir, keystore.StandardScryptN, keystore.StandardScryptP)
+	account, err := ks.NewAccount(password)
+	if err != nil {
+		return errors.Wrap(err, "failed to create keystore")
+	}
+	fmt.Printf("Created account: %s", account.Address.Hex())
+
+	return nil
+}
+
+// UpdateClientEthereumAddress updates Ethereum address
+func UpdateClientEthereumAddress(homedir, address string) (err error) {
+	configFile := path.Join(homedir, ConfigFile)
+	buf, err := os.ReadFile(configFile)
+	if err != nil {
+		return err
+	}
+	cfg := &Configuration{}
+	err = yaml.Unmarshal(buf, cfg)
+	if err != nil {
+		return err
+	}
+
+	cfg.WalletAddress = address
+
+	text, err := yaml.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(configFile, text, 0644)
+
+	return err
+}
+
+// ImportAccount imports account using mnemonic
+// password is used to lock and unlock keystorage before signing transaction
+func ImportAccount(homedir, mnemonic, password string) (string, error) {
+	// 1. Create storage and account if it doesn't exist and add account to it
+	keyDir := path.Join(homedir, WalletDir)
+	ks := keystore.NewKeyStore(keyDir, keystore.StandardScryptN, keystore.StandardScryptP)
+
+	// 2. Init wallet
+	wallet, err := hdw.NewFromMnemonic(mnemonic)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to import from mnemonic")
+	}
+
+	pathD := hdw.MustParseDerivationPath("m/44'/60'/0'/0/0")
```

Comment:
Comments the bip44 wallet path params or links to the bip44 wiki page so that caller understand what does that mean.

Reply:
Fixed in 432ae2654218b5007d6fe8ac1aefdbc28bd67c1f